packages/destination-actions/src/destinations/amazon-conversions-api/trackConversion/utils.ts

```diff
@@ -0,0 +1,372 @@
+    }
+    if (hasStringValue(amznAdStorage) && hasStringValue(amznUserData)) {
+        consentData.amazonConsent = {
+            amznAdStorage,
+            amznUserData
+        } as AmazonConsentFormat
+    }
+    if (hasStringValue(tcf)) {
+        consentData.tcf = tcf
+    }
+    if (hasStringValue(gpp)) {
+        consentData.gpp = gpp
+    }
+
+    return consentData as ConsentData
+}
+
+/**
+ * General normalization utility function for string values
+ * @param value - Input string to normalize
+ * @param allowedChars - RegExp pattern of characters to remove
+ * @param trim - Whether to trim whitespace (default: true)
+ * @returns Normalized string
+ */
+export function normalize(value: string, allowedChars: RegExp, trim = true): string {
+    let normalized = value.toLowerCase().replace(allowedChars, '')
+    if (trim) normalized = normalized.trim()
+    return normalized
+}
+
+// RegExp patterns for normalization
+const alphanumeric = /[^a-z0-9]/g
+const emailAllowed = /[^a-z0-9.@+\-]/g
+const nonDigits = /[^\d]/g
+const whitespace = /\s+/g
+
+/**
+ * Normalizes an email address according to Amazon's requirements:
+ * Lowercase, remove all non-alphanumeric characters except [.@-],
+ * and remove any leading or trailing whitespace.
+ */
+export function normalizeEmail(email: string): string {
+    return normalize(email, emailAllowed)
+}
+
+/**
+ * Normalizes a phone number by removing all non-digit characters
+ */
+export function normalizePhone(phone: string): string {
+    return normalize(phone, nonDigits)
+}
+
+/**
+ * Normalizes a standard string by lowercasing and removing non-alphanumeric characters
+ * Used for firstName, lastName, address, city, state
+ */
+export function normalizeStandard(value: string): string {
+    return normalize(value, alphanumeric)
+}
+
+/**
+ * Normalizes a postal code by removing spaces
+ */
+export function normalizePostal(postal: string): string {
+    return normalize(postal, whitespace, false)
+}
+
+/**
+ * Helper function to smart-hash a value with proper normalization
+ * @param value - The value to hash
+ * @param normalizeFunction - Optional normalization function to apply
+ * @returns The properly normalized and hashed value, or original if already hashed
+ */
+export function smartHash(value: string, normalizeFunction?: (value: string) => string): string {
+    return processHashing(value, 'sha256', 'hex', normalizeFunction)
+}
+
+/**
+ * Sends event data to the Amazon Conversions API
+ *
+ * @param request The request client
+ * @param settings The API settings
+ * @param eventData The event data to send (single event or array of events)
+ * @param throwHttpErrors Whether to throw HTTP errors (defaults to false)
+ * @returns The API response with ImportConversionEventsResponse data
+ */
+export async function sendEventsRequest(
+    request: RequestClient,
+    settings: Settings,
+    eventData: EventData | EventData[],
+    throwHttpErrors: boolean = false
```

Comment:
Linter complaining that type definition not needed because it's inferred. 

```suggestion
    throwHttpErrors = false
```